prov/tcp/src/tcpx_attr.c

```diff
@@ -75,7 +75,7 @@ static struct fi_ep_attr tcpx_ep_attr = {
 	.type = FI_EP_MSG,
 	.protocol = FI_PROTO_SOCK_TCP,
 	.protocol_version = 0,
-	.max_msg_size = SIZE_MAX,
+	.max_msg_size = OFI_MAX_SOCKET_BUF_SIZE,
```

Comment:
This should be reduced by the size of the tcp header.

Reply:
Hmm... wait on this.  The tcp protocol should be able to support u64, and the code can handle socket sending/receiving fewer bytes than requested.  So, SIZE_MAX should be okay.  I think we just need to adjust the actual socket calls on windows to only request INT_MAX bytes at a time.  The upper level code should handle this and continue sending/receiving the entire transfer.

Reply:
And my last comment only applies if the socket has been set to nonblocking.  I think that's always the case, but we might want to add a comment above send/recv to indicate this.